src/components/application_manager/src/request_controller.cc

```diff
@@ -190,8 +190,9 @@ RequestController::TResult RequestController::addHMIRequest(
 
   const uint64_t timeout_in_mseconds =
       static_cast<uint64_t>(request->default_timeout());
-  RequestInfoPtr request_info_ptr(
-      new HMIRequestInfo(request, timeout_in_mseconds));
+  RequestInfoPtr request_info_ptr =
+      utils::MakeShared<HMIRequestInfo>(request, timeout_in_mseconds);
+  request_info_ptr->set_hmi_level(mobile_apis::HMILevel::INVALID_ENUM);
```

Comment:
@vlantonov for me it looks like we have to set `INVALID_ENUM` to the `HMIRequestInfo` in constructor. As result you can avoid additional call to `set_hmi_level` function